src/trace/cursor/mod.rs

```diff
@@ -113,6 +113,28 @@ pub trait Cursor<'s, K, V, T, R> {
     fn rewind_vals(&mut self);
 }
 
+pub trait Consumer<K, V, R> {
+    type ValueConsumer<'a>: ValueConsumer<'a, V, R>
+    where
+        Self: 'a;
+
+    fn key_valid(&self) -> bool;
+
+    fn next_key(&mut self) -> (K, Self::ValueConsumer<'_>);
+
+    fn seek_key(&mut self, key: &K)
+    where
+        K: Ord;
+}
+
+pub trait ValueConsumer<'a, V, R> {
+    fn value_valid(&self) -> bool;
```

Comment:
Let's call this `val_valid` for consistency with `Cursor`.

Reply:
I personally dislike the shortened names to be honest, that's why I changed it